src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -688,7 +688,7 @@ protected virtual void MarkUserDependency (MemberReference context, CustomAttrib
 				assembly = _context.GetLoadedAssembly (assemblyName);
 				if (assembly == null) {
 					_context.LogWarning (
-						$"Could not resolve '{assemblyName}' assembly dependency specified in a `PreserveDependency` attribute that targets method '{context.FullName}'", 2003, context.Resolve ());
+						$"Could not resolve '{assemblyName}' assembly dependency specified in a `PreserveDependency` attribute that targets method '{(context as MethodReference).GetDisplayName ()}'", 2003, context.Resolve ());
```

Comment:
This is potentially dangerous - the cast may return null in which case this will throw.
I think we should have an extension method for `MemberReference`, which for now will handle methods nicely and everthing else as `FullName`...